From d866ef4995568e6643daba51eb08d1a87871611c Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@videolabs.io>
Date: Fri, 6 Jan 2017 14:11:26 +0100
Subject: [PATCH 20/39] d3d11: include initguid.h before d3d11.h so we have all
 the class IIDs needed

Directly from the headers

--
replaces https://patches.videolan.org/patch/15371/
* no need for initguid.h in common.h
---
 modules/codec/avcodec/d3d11va.c         | 6 +-----
 modules/video_output/win32/direct3d11.c | 2 +-
 2 files changed, 2 insertions(+), 6 deletions(-)

diff --git a/modules/codec/avcodec/d3d11va.c b/modules/codec/avcodec/d3d11va.c
index 0bb2ffd..53ddd30 100644
--- a/modules/codec/avcodec/d3d11va.c
+++ b/modules/codec/avcodec/d3d11va.c
@@ -48,7 +48,7 @@
 #include "directx_va.h"
 
 #define COBJMACROS
-#define INITGUID
+#include <initguid.h>
 #include <d3d11.h>
 #include <libavcodec/d3d11va.h>
 
@@ -94,10 +94,6 @@ vlc_module_end()
 #endif
 
 DEFINE_GUID(GUID_CONTEXT_MUTEX,      0x472e8835, 0x3f8e, 0x4f93, 0xa0, 0xcb, 0x25, 0x79, 0x77, 0x6c, 0xed, 0x86);
-DEFINE_GUID(IID_ID3D11VideoDevice,   0x10EC4D5B, 0x975A, 0x4689, 0xB9, 0xE4, 0xD0, 0xAA, 0xC3, 0x0F, 0xE3, 0x33);
-DEFINE_GUID(IID_ID3D11VideoContext,  0x61F21C45, 0x3C0E, 0x4a74, 0x9C, 0xEA, 0x67, 0x10, 0x0D, 0x9A, 0xD5, 0xE4);
-DEFINE_GUID(IID_IDXGIDevice,         0x54ec77fa, 0x1377, 0x44e6, 0x8c, 0x32, 0x88, 0xfd, 0x5f, 0x44, 0xc8, 0x4c);
-DEFINE_GUID(IID_ID3D10Multithread,   0x9b7e4e00, 0x342c, 0x4106, 0xa1, 0x9f, 0x4f, 0x27, 0x04, 0xf6, 0x89, 0xf0);
 
 DEFINE_GUID(DXVA_Intel_H264_NoFGT_ClearVideo,       0x604F8E68, 0x4951, 0x4c54, 0x88, 0xFE, 0xAB, 0xD2, 0x5C, 0x15, 0xB3, 0xD6);
 
diff --git a/modules/video_output/win32/direct3d11.c b/modules/video_output/win32/direct3d11.c
index cb59794..7b87255 100644
--- a/modules/video_output/win32/direct3d11.c
+++ b/modules/video_output/win32/direct3d11.c
@@ -38,7 +38,7 @@
 #include <math.h>
 
 #define COBJMACROS
-#define INITGUID
+#include <initguid.h>
 #include <d3d11.h>
 
 /* avoided until we can pass ISwapchainPanel without c++/cx mode
-- 
2.10.1.windows.1

